### PR TITLE
Feature/std ranges part9

### PR DESCRIPTION
**Pull request by whaeck**
Pulling some more things from NanoRange:
- the mismatch algorithm
- counted_view, filter_view, keys and values views, split_view

All the tests compile on intel-classic/2022.0, intel-classic/2023.1.

### CODE REVIEW

File: src/tools/std20/algorithm/test/mismatch.test.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+// what we are testing
+#include "tools/std20/algorithm.hpp"
+
+// other includes
+
+// convenience typedefs
+//using namespace njoy::tools;
+namespace std20 = nano;
+
+SCENARIO( "equal" ) {
```
Comment (mswwilson) [inline src/tools/std20/algorithm/test/mismatch.test.cpp:13]: "equal" -> "mismatch"

File: src/tools/std20/views/test/counted.test.cpp [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+#include <forward_list>
+#include <list>
+#include <vector>
+#include "tools/std20/algorithm.hpp"
+
+// convenience typedefs
+//using namespace njoy::tools;
+namespace std20 = nano;
+
+SCENARIO( "take_view" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/counted.test.cpp:17]: "take_view" -> "counted"

File: src/tools/std20/views/test/counted.test.cpp [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+		    CHECK( std20::ranges::view< Range > );
+		    CHECK( std20::ranges::sized_range< Range > );
+        CHECK( std20::ranges::forward_range< Range > );
+        CHECK( std20::ranges::bidirectional_range< Range > );
+		    CHECK( std20::ranges::random_access_range< Range > );
+		    CHECK( ! std20::ranges::contiguous_range< Range > );
+		    CHECK( std20::ranges::common_range< Range > );
+      }
+
+      THEN( "the take_view range and iterator associated types are correct" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/counted.test.cpp:164]: "take_view" -> "counted" 

Likewise for other THENs in scenario

File: src/tools/std20/views/test/filter.test.cpp [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+		    CHECK( std20::ranges::view< Range > );
+		    CHECK( ! std20::ranges::sized_range< Range > );
+        CHECK( std20::ranges::forward_range< Range > );
+        CHECK( ! std20::ranges::bidirectional_range< Range > );
+		    CHECK( ! std20::ranges::random_access_range< Range > );
+		    CHECK( ! std20::ranges::contiguous_range< Range > );
+		    CHECK( std20::ranges::common_range< Range > );
+      }
+
+      THEN( "the take_view range and iterator associated types are correct" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/filter.test.cpp:46]: "take_view" -> "filter" 

Likewise for other THENs in scenario

File: src/tools/std20/views/test/reverse.test.cpp [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+#include <forward_list>
+#include <list>
+#include <vector>
+#include "tools/std20/algorithm.hpp"
+
+// convenience typedefs
+//using namespace njoy::tools;
+namespace std20 = nano;
+
+SCENARIO( "subrange" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/reverse.test.cpp:17]: "subrange" -> "reverse"

File: src/tools/std20/views/test/reverse.test.cpp [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+    std::list< int > values = { -2, -1, 0, 1, 2 };
+
+    WHEN( "when iterators are used" ) {
+
+      auto chunk = values | std20::views::reverse;
+      using Range = decltype(chunk);
+      using Iterator = nano::iterator_t< Range >;
+
+      THEN( "the subrange satisfies the required concepts" ) {
```
Comment (mswwilson) [inline src/tools/std20/views/test/reverse.test.cpp:32]: "subrange" -> "reverse" 

Likewise for other THENs in scenario

**Review by mswwilson — DISMISSED** [review]

**Review by mswwilson — APPROVED** [review]

File: src/tools/std20/algorithm/test/mismatch.test.cpp (hunk 1 above)
Comment (whaeck) [reply src/tools/std20/algorithm/test/mismatch.test.cpp:13]: fixed in fix/review

File: src/tools/std20/views/test/counted.test.cpp (hunk 2 above)
Comment (whaeck) [reply src/tools/std20/views/test/counted.test.cpp:17]: same

File: src/tools/std20/views/test/counted.test.cpp (hunk 3 above)
Comment (whaeck) [reply src/tools/std20/views/test/counted.test.cpp:164]: same

File: src/tools/std20/views/test/filter.test.cpp (hunk 4 above)
Comment (whaeck) [reply src/tools/std20/views/test/filter.test.cpp:46]: same

File: src/tools/std20/views/test/reverse.test.cpp (hunk 5 above)
Comment (whaeck) [reply src/tools/std20/views/test/reverse.test.cpp:17]: same

File: src/tools/std20/views/test/reverse.test.cpp (hunk 6 above)
Comment (whaeck) [reply src/tools/std20/views/test/reverse.test.cpp:32]: same

**Pull request merged**